script.js

```diff
@@ -1,4 +1,398 @@
+
+/*********************************************************************************************************/
+// 4. Shuffle the Deck.
+
+var shuffledDeck = function () {
+  for (i = 0; i < deck.length; i++) {
+    // Scrambling the order so that the array can shuffled.
+    let j = Math.floor(Math.random() * deck.length);
+    let temp = deck[i];
+    deck[i] = deck[j];
+    deck[j] = temp;
+  }
+  for (let i = 0; i < deck.length; i++) {
+    let cardImage = document.createElement("img");
+    let card = deck[i];
+    cardImage.src = "./card/" + card + ".png";
+    document.getElementById("shuffledCardDeck").append(cardImage);
+  }
+};
+
+getShuffledDeck = shuffledDeck();
+console.log("Shuffled Card Deck Array:");
+console.log(deck);
+/*********************************************************************************************************/
+// 5. Display shuffled the Deck.
+var displayShuffledDeck = function () {
+  myOutputValue = `Shuffled Deck:` + `<br>` + `${deck}`;
   return myOutputValue;
 };
+
+getDisplayOfShuffledDeck = displayShuffledDeck();
+
+/*********************************************************************************************************/
+// 6. Getting the Value on the Card
+var cardValue = function (card) {
+  let data = card.split("-"); // Card name is label as value . suits. value will be in the first index of the array.
+  console.log(data);
+  let value = data[0];
+  console.log("Value of Card:");
+  console.log(value);
+
+  // To check if the value contain digit or else if it is not a number -> A can be 10 or 11 and if (K,Q,J) -> 10.
+  if (isNaN(value)) {
+    if (value == "A") {
+      return 11;
+    }
+    return 10; // -> For J,Q,K.
+  }
+  return parseInt(value); // to return the value of a card if it is a number.
+};
+
+/*********************************************************************************************************/
+// 7. To check hidden card value.
+hidden = deck.pop(); // last card is designated as the hidden card.
+getHiddenCardValue = cardValue(hidden);
+console.log("Card:");
+console.log(hidden);
+console.log("Value of Hidden Card:");
+console.log(getHiddenCardValue);
+
+/*********************************************************************************************************/
+// 8. To refine the value of Ace Card
+var AceCardValue = function (card) {
+  if (card[0] == "A") {
+    return 1;
+  }
+  return 0;
+};
+
+/*********************************************************************************************************/
+// 9. To track the total count for dealer.
+getAceCardValueForHidden = AceCardValue(hidden);
+
+dealerTotal += getHiddenCardValue;
+dealerAceCount += getAceCardValueForHidden;
+
+console.log("Dealer Score:");
+console.log(dealerTotal, dealerAceCount);
+
+/*********************************************************************************************************/
+// 10. To give dealer another card when the score is below 17.
+var dealerDrawCard = function () {
```

Comment:
isit possible to combine both dealer and player draw card into one function `drawCard(player/dealer)` and then pass in `player` or `dealer` into parameter to be executed ?